runtime/src/accounts_db.rs

```diff
@@ -1347,19 +1423,38 @@ impl AccountsDB {
         dead_slots
     }
 
-    fn clean_dead_slots(&self, dead_slots: &mut HashSet<Slot>) {
+    pub fn clean_dead_slots(&self, dead_slots: &HashSet<Slot>) {
         if !dead_slots.is_empty() {
             {
                 let mut measure = Measure::start("clean_dead_slots-ms");
-                let index = self.accounts_index.read().unwrap();
                 let storage = self.storage.read().unwrap();
+                let mut stores: Vec<Arc<AccountStorageEntry>> = vec![];
                 for slot in dead_slots.iter() {
                     for store in storage.0.get(slot).unwrap().values() {
-                        for account in store.accounts.accounts(0) {
-                            index.unref_from_storage(&account.meta.pubkey);
-                        }
+                        stores.push(store.clone());
                     }
                 }
+                drop(storage);
+                datapoint_info!("clean_dead_slots", ("stores", stores.len(), i64));
+                let pubkeys: Vec<Pubkey> = {
+                    self.thread_pool_clean.install(|| {
+                        stores
+                            .into_par_iter()
+                            .map(|store| {
+                                let accounts = store.accounts.accounts(0);
+                                accounts
+                                    .into_iter()
+                                    .map(|account| account.meta.pubkey)
+                                    .collect::<Vec<Pubkey>>()
+                            })
+                            .flatten()
```

Comment:
Is `flatten()`-ing here really needed? I think [a double-nesting loop](https://github.com/solana-labs/solana/pull/8799/files#diff-2099c5256db4eb5975c8834af38f6456R1456) could avoid potentially heavy memory ops here?